docs/examples/09-graph_communication.jl

```diff
@@ -41,85 +41,49 @@ end
 source = Cint[rank]
 graph_comm = MPI.Dist_graph_create(comm, source, degree, dest)
 
+# Query number of ranks that point to this rank, and number of ranks this rank point to
+indegree, outdegree, _ = MPI.Dist_graph_neighbors_count(graph_comm)
+
+# Query which ranks that point to this rank, and which ranks this rank point to
+inranks = Vector{Cint}(undef, indegree)
+outranks = Vector{Cint}(undef, outdegree)
+MPI.Dist_graph_neighbors!(graph_comm, inranks, outranks)
```

Comment:
We need to make an allocating version of this function!

Reply:
You mean a function `Dist_graph_neighbors` that combine `Dist_graph_neighbors_count` and `Dist_graph_neighbors!`? Should it always return weights then, where weights are set to 1 for unweighted graphs? Or perhaps set to `nothing`?

Reply:
I think returning `nothing` would make sense